File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/storage/hamburgesa/HamburgesaStorageServiceXML.java
package storage.hamburgesa;

import config.ConfigApp;
import dto.hamburgesa.ListaHamburgesaDto;
import dto.ingrediente.ListaIngredienteDto;
import mapper.HamburgesaMapper;
import models.Hamburgesa;
import org.simpleframework.xml.core.Persister;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class HamburgesaStorageServiceXML implements HamburgesaStorageService{

    ConfigApp configApp = ConfigApp.getInstance();
    File file = new File(configApp.getAPP_DATA()+File.separator+"haburgesa.xml");

    public HamburgesaStorageServiceXML() throws IOException {}

    Persister persister = new Persister();

    @Override
    public void saveAll(List<Hamburgesa> entities) throws Exception {
        try {
            persister.write(HamburgesaMapper.toDto(entities), file);
        }catch (Exception e){
            System.err.println(e.getMessage());
        }
    }

    @Override
    public List<Hamburgesa> loadAll() throws Exception {
        List<Hamburgesa> lista = new ArrayList<>();
        if(!file.exists()) return lista;
        try{
            lista = HamburgesaMapper.toHamburgesas(persister.read(ListaHamburgesaDto.class, file));
        }catch (Exception e){
            System.err.println(e.getMessage());
        }
        return lista;
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/storageService/Hamburger/HamburgesaStorageServiceXML.kt
package storageService.Hamburger

import config.ConfigApp
import mapper.toDto
import mapper.toHamburgesa
import model.Hamburgesa
import mu.KotlinLogging
import org.simpleframework.xml.core.Persister
import java.io.File

class HamburgesaStorageServiceXML : HamburgesaStorageService {

    private val config = ConfigApp
    private val file = File(config.APP_DATA+ File.separator+"hamburgesa.xml")
    private val logger = KotlinLogging.logger {  }

    private val serializer = Persister()

    override fun saveAll(entities: List<Hamburgesa>) {
        logger.debug { "Se guardan todos las hamburgesas en el archivo XML." }
        serializer.write(entities.toDto(), file)
    }

    override fun loadAll(): List<Hamburgesa> {
        logger.debug { "Se cargan todos las hamburgesas del archivo XML." }
        if(!file.exists()) return emptyList()
        return serializer.read(dto.hamburgesa.ListaHamburgesasDto::class.java, file).toHamburgesa()
    }
}